spec/install/gemfile/eval_gemfile_spec.rb

```diff
@@ -41,6 +41,13 @@
       bundle! :install
       expect(the_bundle).to include_gem("a 1.0")
     end
+
+    # Make sure that we are properly comparing path based gems between the
+    # parsed lockfile and the evaluated gemfile.
```

Comment:
For this one, it will fail in a blaze of glory (when using the old code) with something along the lines of the following outputted (trimmed down a bit for readability):

```console
$ ruby -Ilib:spec -rsupport/hax -rsupport/artifice/fail exe/bundle install --no-color
Resolving dependencies...
Using a 1.0 from source at `../gems/a`
Using bundler 1.15.1
Bundle complete! 1 Gemfile dependency, 2 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
# $? => 0

$ ruby -Ilib:spec -rsupport/hax -rsupport/artifice/fail exe/bundle install --deployment --no-color
You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated Gemfile.lock to version control.

The list of sources changed
The gemspecs for path gems changed

You have added to the Gemfile:
* source: source at `../gems/a`
# $? => 16
```

Not sure if it was worth adding all this as a comment to the spec, but leaving this hear for some context